_docs/other/developer-guide.md

```diff
@@ -3,56 +3,71 @@ title: Developer Guide
+1. Clone the repo to your computer.
+2. Make sure that you have Java 21 installed.
+3. Navigate to the Geyser root directory and run `git submodule update --init --recursive`. This command downloads all the needed submodules for Geyser and is a crucial step in this process.
+4. Run `./gradlew build` and locate to `bootstrap/<platform>/build/` folder.
 
 ## Project layout
+
+Geyser's code is divided into different modules. For example:
+
 * `bootstrap` is where we hold the specific platform code. So if you're porting Geyser to a new platform, or working with platform-specific code, you likely want to be in here.
-* `core` is where connections are handled and the data conversion is done. The majority of Geyser work is held here.
+* `core` is where connections are handled and the data/packet conversion is done. The majority of Geyser work is held here.
+* `api` is where the Geyser API is found.
+* `build-logic` is where the Kotlin Gradle plugins used to build Geyser are.
+
+## Compiler/IDE Tools
 
-# Compiler Tools
+### Minecraft Development (IntelliJ IDEA plugin)
 
-## Lombok
+The [Minecraft Development](https://github.com/minecraft-dev/MinecraftDev) IntelliJ IDEA plugin is an awesome tool that can
+be very helpful when working on Geyser. It allows you to view `.nbt` files, easily link mixins to the source they modify, and much 
+more. While not strictly needed, it is definitely recommended.
+
+### Lombok
 
 *Note that the latest version of IntelliJ IDEA does not require additional setup in order to develop with Geyser.*
 
-If you're using an IDE for editing any of the GeyserMC projects you will most likely need to install the Project Lombok plugin as it is used to generate a bunch of handy functions. You can edit without it but you may get missing functions and or other issues displayed in your IDE. Please see the IDE section on their site for the supported plugins and install methods [https://projectlombok.org/setup/overview](https://projectlombok.org/setup/overview).
+If you're using an IDE for editing any of the GeyserMC projects you will most likely need to install the Project Lombok plugin as it is used to generate a bunch of handy functions. 
+You can edit without it but you may get missing functions and or other issues displayed in your IDE. Please see the IDE section on their site for the supported plugins and install methods [https://projectlombok.org/setup/overview](https://projectlombok.org/setup/overview).
+
+### 
 
-# Protocol Information
+## Protocol Information
 
-## GopherTunnel
+### GopherTunnel
 [GopherTunnel](https://github.com/Sandertv/gophertunnel/tree/master/minecraft/protocol/packet) is a library for Bedrock Edition written in Go. The source code is an excellent documentation for the Bedrock protocol.
 
-## wiki.vg
+### wiki.vg
 For a full rundown of the Java Edition protocol, see [here](https://wiki.vg/Protocol).
 
-## wiki.vg (Bedrock)
+### wiki.vg (Bedrock)
```

Comment:
Does this need a separate header?

Reply:
All programs/pages have a header at the moment, just kept it - added another `#` as that's less "blown up"
Current page:
![image](https://github.com/GeyserMC/GeyserWiki/assets/105284508/075c3f3e-6776-4ebc-a074-4e01e2855a79)
Latest changes:
![image](https://github.com/GeyserMC/GeyserWiki/assets/105284508/9c2da441-383a-4397-8188-67591f896144)

Reply:
I meant like, wiki.vg is one program with distinct links. Can't it just be like

**Java** - blah<br>
**Bedrock** - blah

Reply:
sure; but that's been there before.. I'm not really looking to rework the development guide here, that's for another PR

Reply:
Fair enough, can handle that later :)